controllers/rollingupgrade_controller.go

```diff
@@ -704,9 +705,43 @@ func (r *RollingUpgradeReconciler) Process(ctx *context.Context,
 		return
 	}
 
+	//Validation step: check if all the nodes have the latest launchconfig.
+	r.info(ruObj, "Validating the launch config of nodes and ASG")
+	if err := r.validateNodesLaunchConfig(ruObj); err != nil {
+		r.error(ruObj, err, "Launch config validation failed")
+		r.finishExecution(StatusError, nodesProcessed, ctx, ruObj)
+		return
+	}
+	r.info(ruObj, "Launch config validation succeeded")
+
 	r.finishExecution(StatusComplete, nodesProcessed, ctx, ruObj)
 }
 
+//Check if ec2Instances and the ASG have same launch config.
+func (r *RollingUpgradeReconciler) validateNodesLaunchConfig(ruObj *upgrademgrv1alpha1.RollingUpgrade) error {
+	//Get ASG launch config
+	err := r.populateAsg(ruObj)
+	if err != nil {
+		return errors.New("Unable to populate the ASG object")
+	}
+	asg, err := r.GetAutoScalingGroup(ruObj.Name)
+	if err != nil {
+		return fmt.Errorf("Unable to load ASG with name: %s", ruObj.Name)
+	}
+	launchDefinition := NewLaunchDefinition(asg)
```

Comment:
LaunchDefinition can have a `launchConfig` or a `launchTemplate` inside it. We'll have to handle both the cases here.

Reply:
Addressed this issue with latest commit.